### PR TITLE
Replace deprecated functions and packages

### PR DESCRIPTION
**Pull request by padlar**
### What this PR does

Replaces the following deprecated function calls -
grpc.WithInsecure() -> grpc.WithTransportCredentials
grpc.WithDialer -> grpc.WithContextDialer
exec.Stream -> exec.StreamWithContext

Replaces the following deprecated packages -
"k8s.io/utils/pointer" ->  "k8s.io/utils/ptr"
"github.com/go-kit/kit/log" -> "github.com/go-kit/log

Related to #11316 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note

```

### CODE REVIEW

File: pkg/util/net/grpc/grpc.go [hunk 1]
```diff
@@ -34,32 +35,33 @@ import (
...
-		grpc.WithInsecure(),
-		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
-			return net.DialTimeout("unix", addr, timeout)
-		}),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(dialer),
 		grpc.WithBlock(), // dial sync in order to catch errors early
 	}
 
+	ctxTimeout := time.Duration(timeout+CONNECT_TIMEOUT_SECONDS) * time.Second
```
Comment (xpivarc) [inline pkg/util/net/grpc/grpc.go:56]: nit: can we rename the `CONNECT_TIMEOUT_SECONDS` to `defaultTimeoutSeconds` or anything similar?

File: tests/testsuite/fixture.go [hunk 2]
```diff
@@ -103,7 +103,6 @@ func SynchronizedBeforeTestSetup() []byte {
 func BeforeTestSuiteSetup(_ []byte) {
 
 	worker := GinkgoParallelProcess()
-	rand.Seed(int64(worker))
```
Comment (xpivarc) [inline tests/testsuite/fixture.go:106]: Is this deprecated?  `"k8s.io/apimachinery/pkg/util/rand"`

File: pkg/virtctl/guestfs/guestfs.go [hunk 3]
```diff
@@ -605,7 +605,7 @@ func createAttacher(client *K8sClient, p *corev1.Pod, command string) error {
 	resChan := make(chan error)
 
 	go func() {
-		resChan <- exec.Stream(remotecommand.StreamOptions{
+		resChan <- exec.StreamWithContext(context.TODO(), remotecommand.StreamOptions{
```
Comment (xpivarc) [inline pkg/virtctl/guestfs/guestfs.go:608]: This commit does more than it says, please separate this into its own contained commit.

**Review by xpivarc — COMMENTED** [review]
Thank you for opening a PR, few remarks below

File: tests/testsuite/fixture.go (hunk 2 above)
Comment (padlar) [reply tests/testsuite/fixture.go]: reverted the patch here

File: pkg/virtctl/guestfs/guestfs.go (hunk 3 above)
Comment (padlar) [reply pkg/virtctl/guestfs/guestfs.go:608]: updated the description to include this change as well - its replacing 2 additional function calls. I could split this PR into 2 - one for pkg and one for function call deprecations?

**Comment by yj-yoo** [comment]
OMG... am I rate???

File: pkg/storage/export/export/export.go [hunk 4]
```diff
@@ -868,8 +868,8 @@ func (ctrl *VMExportController) createExporterPodManifest(vmExport *exportv1.Vir
 	podManifest.Labels = map[string]string{exportServiceLabel: vmExport.Name}
 	podManifest.Annotations = map[string]string{annCertParams: scp}
 	podManifest.Spec.SecurityContext = &corev1.PodSecurityContext{
-		RunAsNonRoot:   pointer.Bool(true),
-		FSGroup:        pointer.Int64Ptr(kvm),
+		RunAsNonRoot:   ptr.To(true),
```
Comment (oshoval) [inline pkg/storage/export/export/export.go:871]: Please see https://github.com/kubevirt/kubevirt/pull/11348
It already handle some of those (or all of them)
It also uses the internal implementation which is preferable

**Comment by EdDev** [comment]
> "k8s.io/utils/pointer" -> "k8s.io/utils/ptr"

https://github.com/kubevirt/kubevirt/pull/11348#pullrequestreview-1900038048

**Pull request closed without merging**